scripts/new-version-pr-generator.js

```diff
@@ -0,0 +1,154 @@
+</details>
+`.trim();
+}
+
+function failedUpdateMessage(repository, version) {
+  return `The automatic branch creation for ${repository} failed. This can be due to many reasons. To resolve this follow these steps:
+
+  1. Checkout "${repository}"
+  2. Get the latest version of master "git checkout master && git pull"
+  3. Create a new branch based on the version that failed "git checkout -b update-polaris-v${version}"
+  4. Update "@shopify/polaris" by running "npx yarn upgrade @shopify/polaris@${version}"
+  5. Add the changed files to git by running "git add package.json yarn.lock"
+  6. Commit the changes with "git commit -m 'Update @shopify/polaris to ${version}'"
+  7. Push the changes with "git push origin update-polaris-v${version}"
+  8. Create a new pull request for the branch in "${repository}"`;
+}
+
+mkdir(sandbox);
+
+const tasks = repositories.map((repository) => {
+  return new Promise((resolve, reject) => {
+    try {
+      const repositoryDirectory = pathResolve(sandbox, repository);
+
+      // Clone the repositories to the sandbox directory
+      Retry(
+        execSync(
+          `git clone --branch ${baseBranch} --single-branch https://${polarisBotToken}@github.com/Shopify/${repository}.git`,
+          {
+            cwd: sandbox,
+            stdio: 'inherit',
+          },
+        ),
+      );
+
+      const commands = [
+        `git checkout -b update-polaris-${releaseVersion}`,
+        'npm config set @shopify:registry https://registry.npmjs.org',
+        `npx yarn@${YARN_VERSION} upgrade @shopify/polaris@${releaseVersion.replace(
+          'v',
+          '',
+        )} --no-progress --ignore-engines``git add package.json yarn.lock`,
+        `${shopifyPolarisBotGitOverride} git commit -m "Update @shopify/polaris to ${releaseVersion}" --author "${polarisBotName} <${polarisBotEmail}>" -m "${releaseVersion}" --allow-empty`,
+        `git push origin update-polaris-${releaseVersion}`,
+        `curl -d '${JSON.stringify({
+          title: `Update @shopify/polaris to ${releaseVersion} 🚀`,
+          body: pullRequestTemplate(repository, releaseVersion),
+          head: `update-polaris-${releaseVersion}`,
+          base: baseBranch,
+        })}' -X POST https://api.github.com/repos/shopify/${repository}/pulls?access_token=${polarisBotToken}`,
+      ];
+
+      // Run the commands in the cloned repository directories
+      commands.forEach((command) => {
+        Retry(
+          execSync(command, {
+            cwd: repositoryDirectory,
+            stdio: 'inherit',
+          }),
+        );
+      });
+
+      resolve(repository);
+    } catch (error) {
+      const errorMessage = `${error}\n\n${failedUpdateMessage(
+        repository,
+        releaseVersion,
+      )}`;
+      reject(errorMessage);
+    }
+  });
+});
+
+Promise.all(tasks)
+  .then((repository) =>
+    console.log(
+      `Done: Pull request made for ${releaseVersion} in ${repository}`,
+    ),
+  )
+  .catch((error) => {
```

Comment:
I'm not entirely sure how `Promise.all` interacts with `reject()`, make sure you test this out

Reply:
From previous experience, reject should go into the the catch. I will test it though, thanks @amrocha 👍